app/javascript/search.js

```diff
@@ -86,3 +86,23 @@ document.addEventListener('turbolinks:load', function() {
     }
   })
 })
+
+document.addEventListener('turbolinks:load', () => {
+  document.getElementById('search').addEventListener('input', event => {
```

Comment:
This listener is being added every time any page loads, is there any way to use event delegation for this?

Reply:
Is this solved now with the guard clause? Still not quite sure I 100% understand how Turbolinks and event listeners interact and how event delegation fixes about it.

Reply:
Event delegation tells the page that you want to listen for this specific event, and if it fires then call my function.  When you assign an event handler inside of a `turbolinks:load` event, every time the load event is called you tell the page you want to listen for this specific event.  Since turbolinks doesn't perform hard refreshes of the page the old event listeners are still intact, essentially having the same listener function be called multiple times.

I'm having a hard time explaining this clearly in text, so if this doesn't clear it up for you I can explain over a voice call a lot easier and with more detail.

Reply:
Ok! That was a fantastic explanation, thanks for that! I think I've fixed this now, let me know if this change solves this issue.

Reply:
Looks good now :+1: